plugin.md

```diff
@@ -2,11 +2,44 @@
 
 ## Writing Plugins
 
+The main method that gets called is `ServeDNS`. It has three parameters:
+
+* a `context.Context`;
+* `dns.ResponseWriter` that is, basically, the client's connection;
+* `*dns.Msg` the request from the client.
+
+`ServeDNS` returns two values, a response code and an error. If the error is not nil CoreDNS
+will return a generic SERVFAIL to the client. The response code tells CoreDNS if a reply has been
+written by the plugin chain or not. In the latter case CoreDNS will take care of that.
+
+CoreDNS treats:
+
+* SERVFAIL (dns.RcodeServerFailure)
+* REFUSED (dns.RcodeRefused)
+* FORMERR (dns.RcodeFormatError)
+* NOTIMP (dns.RcodeNotImplemented)
+
+as special and will then assume *nothing* has written to the client. In all other cases it assumes
```

Comment:
s/has written/has been written/

(BTW, it's John's fault - his original comment should have been `s/has written/has been written/g`   ;-) )